extensions/wikia/Helios/User.class.php

```diff
@@ -218,6 +221,14 @@ public static function wasResetPassAuth( $username, $password ) {
 		return $authMethod == self::AUTH_TYPE_RESET_PW;
 	}
 
+	public static function checkAuthenticationStatus( $username, $password, $status ) {
+		if ( empty( self::$authenticationCache[$username][$password][self::STATUS_NAME] ) ) {
+			return false;
+		}
+
+		return self::$authenticationCache[$username][$password][self::STATUS_NAME] == $status;
```

Comment:
I'm not sure exactly but from my debugging sessions it seems that [`authenticate`](https://github.com/Wikia/app/blob/dev/extensions/wikia/Helios/User.class.php#L141) is called multiple times during a request, so the cache is to prevent mediawiki from making multiple helios calls. This is pretty hacky, but for now I think it's necessary because we haven't fully unwound the authentication flow.